### PR TITLE
Fix: add test coverage

### LINKED ISSUE #97: Enh: Setup code cov to test coverage
This is a straight forward task to add code cov to our workflow so each time a new thing is added we can add tests. and slowly work on coverage for easier refactoring. 

- [ ] add codecov as a dep in hatch tests envt
- [ ] add running codecov push to the CI build
- [ ] Setup codecov service to run on each pr (this could be a separate activity / issue)  - [example of doing that here](https://github.com/stravalib/stravalib/blob/main/.github/workflows/build-test.yml#L59)

### PR DESCRIPTION
**Pull request by lwasser**
closes #97 

Setup codecov. for some reason i can generate an in terminal report but rendered reports are failing for me locally.  but all of the other pieces are setup for this to work

### CODE REVIEW

File: .github/workflows/run-tests.yml [hunk 1]
```diff
@@ -20,4 +20,9 @@ jobs:
       - name: Run tests
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        run: hatch run test:run-tests
+        run: hatch run test:run-coverage
+      - name: Upload coverage reports to Codecov
+        uses: codecov/codecov-action@v4.0.1
```
Comment (willingc) [inline .github/workflows/run-tests.yml:25]:
```suggestion
        uses: codecov/codecov-action@v4.0.2
```

**Review by willingc — COMMENTED** [review]
codecov action just released to 4.0.2

**Comment by lwasser** [comment]
yay it's working ✨ thanks for the cod cov update  @willingc let's merge this puppy and soon we will be on en route towards fearless refactoring! 🚀 

i keep forgetting that you can setup various hatch scripts and i have a few options setup for us to use. actually before i merge i'll document that

File: development.md [hunk 2]
```diff
@@ -6,12 +6,31 @@ pyosMeta uses hatch and hatchling as it's build back end.
...
+   `hatch run test:run-coverage`
+
+3. To run tests with an xml report generated use:
+   `hatch run test:run-report`
+
+   The hatch run-report script is the script used in our CI tests action.
+
+### Modify test scripts
+
+To modify how scripts ar run, you can look at this section in our
```
Comment (willingc) [inline development.md:28]:
```suggestion
To modify how scripts are run, you can look at this section in our
```

**Review by willingc — APPROVED** [review]

**Comment by willingc** [comment]
Excellent work @lwasser :tada: I'm merging. If there is further doc stuff, let's do that in a new PR. 🐶

**Pull request merged**